### PR TITLE
Implement commitAppend

### PR DESCRIPTION
**Pull request by JinnyYi**
Implement commitAppend in ref: https://github.com/aos-dev/go-storage/issues/529.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.14
 require (
 	github.com/aliyun/aliyun-oss-go-sdk v2.1.8+incompatible
 	github.com/aos-dev/go-integration-test/v3 v3.0.0
-	github.com/aos-dev/go-storage/v3 v3.6.0
+	github.com/aos-dev/go-storage/v3 v3.6.1-0.20210427074731-1b3b4e86c950
```
Comment (Xuanwo) [inline go.mod:8]: Please provide `append` related metadata in obejct.

File: storage.go [hunk 2]
```diff
@@ -252,14 +290,8 @@ func (s *Storage) writeAppend(ctx context.Context, o *Object, r io.Reader, size
...
 	options = append(options, oss.ContentLength(size))
-	if opt.HasServerSideEncryption {
-		options = append(options, oss.ServerSideEncryption(opt.ServerSideEncryption))
-	}
-	if opt.HasServerSideDataEncryption {
-		options = append(options, oss.ServerSideDataEncryption(opt.ServerSideDataEncryption))
-	}
-	if opt.HasServerSideEncryptionKeyID {
-		options = append(options, oss.ServerSideEncryptionKeyID(opt.ServerSideEncryptionKeyID))
+	if opt.HasContentMd5 {
```
Comment (Xuanwo) [inline storage.go:293]: We should pass the `content_type` the first time of the `AppendObject` call.

As we added the `content_type` in Object, but that value is not stored by service. If the user tries to `Stat` an object, he/she will found those metadata is missing.

File: service.toml [hunk 3]
```diff
@@ -22,8 +22,11 @@ optional = ["offset", "io_callback", "size"]
 [namespace.storage.op.write]
 optional = ["content_md5", "content_type", "io_callback", "storage_class", "server_side_encryption", "server_side_data_encryption", "server_side_encryption_key_id"]
 
+[namespace.storage.op.create_append]
+optional = ["content_type", "server_side_encryption", "server_side_encryption_key_id", "server_side_data_encryption", "storage_class"]
+
 [namespace.storage.op.write_append]
-optional = ["server_side_encryption", "server_side_data_encryption", "server_side_encryption_key_id"]
+optional = ["content_md5", "io_callback"]
```
[the remark this answers is not in the document]
Comment (JinnyYi) [reply service.toml:29]: It's allowed to set SSE related request headers when position position=0, refer to https://help.aliyun.com/document_detail/31981.html. These headers has been set in `create_append` through append 0 byte to object.

[the remark this answers is not in the document]
Comment (JinnyYi) [reply service.toml:29]: It's allowed to set SSE related request headers when position position=0 referring to https://help.aliyun.com/document_detail/31981.html. These headers have been set in create_append through append 0 byte to object.

File: storage.go (hunk 2 above)
Comment (JinnyYi) [reply storage.go:293]: `content-type` has set in `create_append`.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**